src/plugins/intel_gna/src/transformations/utils/transformation_helper.cpp

```diff
@@ -153,7 +153,8 @@ void remove_single_input_node(std::shared_ptr<ov::Node> node) {
     if (!node_parent) {
         THROW_GNA_EXCEPTION << "The removing node has no parrent node";
     }
-    if (!std::equal(input_node_shape.begin(), input_node_shape.end(), output_node_shape.begin())) {
+    if (input_node_shape.size() != output_node_shape.size() ||
```

Comment:
With this change I think we should rename the function to remove_single_input_output_node.

Reply:
I don't see any huge changes. We just befor std::equal additionally check if compared vectors has different sizes. It is needed for Windows build - Windows STL throws unneeded exception in a such case

Reply:
I didn't say there are huge changes, but that with these changes it makes sense to rename the function, because it's what it actually does. Anyway now it's really hard to follow this and other PRs, because they are interdependent and the code is quickly outdated.